src/interp/interp.cc

```diff
@@ -260,31 +268,46 @@ void Store::Collect() {
     thread->Mark();
   }
 
-  // TODO: better GC algo.
-  // Loop through all newly marked objects and mark their referents.
-  std::vector<bool> all_marked(object_count, false);
-  bool new_marked;
-  do {
-    new_marked = false;
-    for (size_t i = 0; i < object_count; ++i) {
-      if (!all_marked[i] && marks_[i]) {
-        all_marked[i] = true;
-        objects_.Get(i)->Mark(*this);
-        new_marked = true;
-      }
-    }
-  } while (new_marked);
+  // This vector is often empty since the default maximum
+  // recursion is usually enough to mark all objects.
+  while (!context.unmarked_objects.empty()) {
+    size_t index = context.unmarked_objects.back();
+
+    assert(context.marks[index]);
+    assert(context.call_depth == 0);
+
+    context.unmarked_objects.pop_back();
```

Comment:
But this line next to the `back()` above since they are kind of part of the same operation?   (I still don't get why we get and pop the pack in single call).

Reply:
I also find it strange, but it seems `pop_back` is a void function:
https://www.cplusplus.com/reference/vector/vector/pop_back/

I would prefer to do the pop and return the popped item in one step if there is an operation for that.